### PR TITLE
RC_711 Revoke credential flow 

### PR DESCRIPTION
**Pull request by Sreejit-K**
The revocation flow already exists where we used to check whether a credential certificatId is stored in the "RevokedCredential" table or not. If the certificateId is found in that table, we would assume that the credential is no longer valid and revoked but the cache is. The current flow is revoking and making the VC undiscoverable. Adding to that, it doesn't have a standalone API for users to leverage. This PR provides an API to revoke a VC and make the entity still discoverable to user but with not digital signature assigned to it.

The new API contract will look like this -
POST /api/v1/{entityName}/{entityId}/revoke

CURL:
curl --location --globoff --request POST '{registry-url}/api/v1/{entityName}/{entityId}/revoke'
--header 'content-type: application/json'
--header 'authorization: Bearer {access-token}'

- It Provides a standalone API for revoking the VC.
- In the existing flow we used to soft delete the VC by marking the status as false and storing the signedData in the RevokedCredential Registry. Which deactivates the entity as a whole, which is not an ideal case. 
- The new flow revokes a credential by updating the _osSignedData field in the Entity table to an Empty String by not marking the status as false and storing the signedData in the RevokedCredential Registry. 
- This basically is removing the signature from the VC rather than the VC itself.

Issue: https://github.com/Sunbird-RC/community/issues/711

### CODE REVIEW

**Comment by tejash-jl** [comment]
instead of modifying the deleteEntityById, use markSignedDataAsNullForAnEntity directly in the revoke controller
instead of modifying existing functionalities
open-closed principle

File: java/registry/src/main/java/dev/sunbirdrc/registry/controller/RegistryEntityController.java [hunk 1]
```diff
@@ -739,4 +739,46 @@ public ResponseEntity<Object> getAttestationCertificate(HttpServletRequest reque
...
+            try {
+                userId = registryHelper.authorize(entityName, entityId, request);
+            } catch (Exception e) {
+                return createUnauthorizedExceptionResponse(e);
+            }
+        }
+        ResponseParams responseParams = new ResponseParams();
+        Response response = new Response(Response.API_ID.REVOKE, "OK", responseParams);
+        try {
+            String tag = "RegistryController.revokeAnExistingCredential" + entityName;
```
Comment (varadeth) [inline java/registry/src/main/java/dev/sunbirdrc/registry/controller/RegistryEntityController.java:762]: add space after revokeAnExistingCredential

File: java/registry/src/main/java/dev/sunbirdrc/registry/controller/RegistryEntityController.java [hunk 2]
```diff
@@ -739,4 +739,46 @@ public ResponseEntity<Object> getAttestationCertificate(HttpServletRequest reque
...
+            watch.start(tag);
+            JsonNode existingEntityNode = getEntityJsonNode(entityName, entityId,false, userId);
+            String SignedData = existingEntityNode.get(OSSystemFields._osSignedData.name()).asText();
```
Comment (varadeth) [inline java/registry/src/main/java/dev/sunbirdrc/registry/controller/RegistryEntityController.java:765]: use camelCase for variable naming. signedData

File: java/registry/src/main/java/dev/sunbirdrc/registry/helper/RegistryHelper.java [hunk 3]
```diff
@@ -16,6 +16,7 @@
 import dev.sunbirdrc.pojos.attestation.Action;
 import dev.sunbirdrc.pojos.attestation.States;
 import dev.sunbirdrc.pojos.attestation.exception.PolicyNotFoundException;
+import dev.sunbirdrc.registry.Constants;
```
Comment (varadeth) [inline java/registry/src/main/java/dev/sunbirdrc/registry/helper/RegistryHelper.java:19]: is this import used?

**Review by varadeth — CHANGES_REQUESTED** [review]

File: java/registry/src/main/java/dev/sunbirdrc/registry/helper/RegistryHelper.java (hunk 3 above)
Comment (Sreejit-K) [reply java/registry/src/main/java/dev/sunbirdrc/registry/helper/RegistryHelper.java:19]: Nope, I have removed might have used it while Writing a separate function in the helper for revoke. But then had to make changes in the flow. So might have missed removing the import. Have removed it now

File: java/registry/src/main/java/dev/sunbirdrc/registry/controller/RegistryEntityController.java (hunk 2 above)
Comment (Sreejit-K) [reply java/registry/src/main/java/dev/sunbirdrc/registry/controller/RegistryEntityController.java]: Have changed it to camelCase. I usually do it but might have missed it here.

File: java/registry/src/main/java/dev/sunbirdrc/registry/controller/RegistryEntityController.java (hunk 1 above)
Comment (Sreejit-K) [reply java/registry/src/main/java/dev/sunbirdrc/registry/controller/RegistryEntityController.java]: Have added the space>

**Review by srprasanna — APPROVED** [review]

**Review by tejash-jl — APPROVED** [review]

**Pull request merged**